react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+
+export default class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      imageIndex: 0,
+      images: this.props.images,
```

Comment:
You can remove this `images` state. Just read from `this.props` if you need the `images`.